docs/contributing-to-analyses/analysis-modules/notebook-structure.md

```diff
@@ -0,0 +1,137 @@
+
+```r
+# Find the repository and module root directories
+repo_root <- rprojroot::find_root(rprojroot::is_git_root)
+module_root <- rprojroot::find_root(rprojroot::is_renv_project)
+```
+
+Alternatively, the module root could be found relative to the repository root, as shown below:
+
+```r
+repo_root <- rprojroot::find_root(rprojroot::is_git_root)
+module_root <- file.path(repo_root, "analyses", "hello-R")
+```
+
+Note that we use `file.path()` to construct paths, which is a platform-independent way to concatenate directories and filenames for safe navigation.
+
+All file paths should be defined as relative paths, with the exception of any input data that might be downloaded as part of the analysis, which may be referenced by a URL.
+
+!!! note
+    When working on an analysis, it is quite common to find that you need a file that you had not anticipated at the start.
+    No problem!
+    Just add that path to the Setup section with the other paths, rather than defining it later in the notebook.
+
+
+### Defining custom functions
+
+Following the Setup section, it is often useful to have a **Functions** section where all custom functions that will be used later in the notebook are defined.
+Keeping the functions in a central place in the notebook makes it easier to find and modify them later, and also makes it easier to reuse them in other analyses.
+Each function should be defined in a separate code chunk, and should be documented with comments to explain what the function does does and how to use it, including inputs and outputs.
+
+In `hello.Rmd`, we define a simple function count cells in a `SingleCellExperiment` object:
+
+```r
+count_sce <- function(sce_file) {
+  # Count cells in an sce file
+  # Args:
+  # - sce_file: path to the SingleCellExperiment file
+  # Returns a data frame with the following columns:
+  # - project_id
+  # - sample_id
+  # - library_id
+  # - n_cells: number of cells in the library
+
+  ...
+}
+```
+
+We generally do _not_ recommend using `source()` to load functions from external files, as this can make it harder to keep track of where functions are defined and can lead to errors if the file is moved or renamed.
+It also means that the notebook is not self-contained, and functions will not be present in the output html files, which can make it harder to share and reproduce the analysis.
+
+
+### Analysis steps
+
+Once all of the functions are defined, the remainder of the notebook should be a series of code chunks that perform the analysis steps, with Markdown text to explain what each step is doing and why.
+Use headings and subheadings as appropriate to break up the analysis into logical sections, and include plots in the notebook to help illustrate the results.
+
+Code chunks should still contain comments as needed to explain logic and implementation, with the Markdown text providing a higher-level overview of the analysis steps, including any interpretation of results.
```

Comment:
ensmallening, and also let's not accidentally imply that comments might not be needed!

```suggestion
Code chunks should still contain comments to explain logic and implementation.
The Markdown text should focus on providing a higher-level overview of the analysis steps, including any interpretation of results.
```